packages/neon-core/__tests__/tx/components/TransactionAttribute.ts

```diff
@@ -123,7 +124,9 @@ describe("deserialize", () => {
     "%s",
     (msg: string, expected: TransactionAttributeLike, data: string) => {
       const result = TransactionAttribute.deserialize(data);
-      expect(result.export()).toEqual(expected);
+      const exported = result.export();

```

Comment:
This is a point of concern. If we `export`, do we expect to get a human-readable format or a transportable format? This might work now because Attributes has been stripped down to just URL. I think we should not be transforming the input unnecessarily unless the user states so. So here we will not transform the input when using the constructor but provide a static method like `TransactionAttribute.Url()` so its clear that you can use a human-readable url there.

Reply:
This is good. 
Just to mention, transaction attribute usage types will probably be extended in future. But the number of types would be little, I think we can do this.

Reply:
Lets assume we had the Remark attribute(which we had previously, leading to the current design). What do you think im sending?

```
var attr = new TransactionAttribute({usage: 1, data: "abcd"});
```

At this point it is unclear if I am attempting to input ASCII or hex. The code currently will assume hex since it looks like hex. This is horrible for the user in terms of design.

Instead, we can go for a static method:
```
var attr = TransactionAttribute.Remark("abcd");
```

We can be clear that this field is always ASCII and no more headaches.

Reply:
Well I already agreed on this.

Reply:
To conclude:
1. constructor will regard user input as hex, a hex checking will be included.
2. Static methods will be used to accept ASCII string.
Still this question is not answered.
> This is a point of concern. If we export, do we expect to get a human-readable format or a transportable format?
I vote for transportable format.

Reply:
Ok then we should change the constructor to accept hex and export to return the hex value for data.

Reply:
Updated